src/app/market/shared/addtocart/add-to-cart.component.ts

```diff
@@ -0,0 +1,34 @@
+import {Component, Input, OnInit} from '@angular/core';
+import { Log } from 'ng2-logger';
+
+import { Listing } from '../../../core/market/api/listing/listing.model';
+import { CartService } from '../../../core/market/api/cart/cart.service';
+import { AddToCartCacheService } from 'app/core/market/market-cache/add-to-cart-cache.service';
+
+@Component({
+  selector: 'app-add-to-cart',
+  templateUrl: './add-to-cart.component.html',
+  styleUrls: ['./add-to-cart.component.scss']
+})
+export class AddToCartComponent implements OnInit {
+
+  private log: any = Log.create('add-to-cart.component id:' + Math.floor((Math.random() * 1000) + 1));
+
+  @Input() listing: Listing;
+
+  constructor(
+    private cartService: CartService,
+    private cache: AddToCartCacheService
```

Comment:
Store the AddToCartCacheService as public variable named 'cache' in the CartService.
Then refer to it as:
this.cartService.cache.isBidded(..)

Reply:
yup 

thanks